testing/file_test/autoupdate.cpp

```diff
@@ -26,20 +25,33 @@ static auto ParseLineNumber(absl::string_view matched_line_number) -> int {
   trimmed = trimmed.trim();
   // NOLINTNEXTLINE(google-runtime-int): API requirement.
   long long val;
-  CARBON_CHECK(!llvm::getAsSignedInteger(trimmed, 10, val));
+  CARBON_CHECK(!llvm::getAsSignedInteger(trimmed, 10, val))
+      << matched_line_number;
   return val;
 }
 
+// A replacement that provides the file and line numbers.
```

Comment:
This comment isn't quite enough for me to figure out what this models. Having read the usage below, perhaps:

```suggestion
// The file and line number that a CHECK line refers to, and the
// replacement from which they were determined, if any.
```

Reply:
Adding to #3214